atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/PersistentLockManager.java

```diff
@@ -62,25 +62,32 @@ public synchronized void shutdown() {
         log.info("Shutdown completed!");
     }
 
-    public synchronized void acquirePersistentLockWithRetry() {
+    // We don't synchronize this method, to avoid deadlocking if {@link #shutdown} is called while attempting
+    // to acquire a lock.
+    public void acquirePersistentLockWithRetry() {
+        while (!tryAcquirePersistentLock()) {
+            waitForRetry();
+        }
+    }
+
+    private synchronized boolean tryAcquirePersistentLock() {
         if (isShutDown) {
```

Comment:
The logic of this class requires access to be synchronized (i.e., we want to ensure that after we set `isShutDown=true`, we never again try to get the lock).

I added `GuardedBy` annotations to enforce this